pandas/core/generic.py

```diff
@@ -1192,125 +1205,94 @@ def _slice(self, slobj, axis=0, kind=None):
-        user will see the error *at the level of setting*
+                removals = set(["__really_unused_name__342424__", "__really_unused_name__xxxxx__", "self"])
+                refs = gc.get_referrers(__really_unused_name__342424__)
 
-        It is technically possible to figure out that we are setting on
-        a copy even WITH a multi-dtyped pandas object. In other words, some blocks
-        may be views while other are not. Currently _is_view will ALWAYS return False
-        for multi-blocks to avoid having to handle this case.
+                names = []
+                for ref in refs:
+                    if inspect.isframe(ref):
+                        for name, __really_unused_name__xxxxx__ in compat.iteritems(ref.f_locals):
+                            if __really_unused_name__xxxxx__ is __really_unused_name__342424__:
+                                names.append(name)
+                    elif isinstance(ref, dict):
+                        for name, __really_unused_name__xxxxx__ in compat.iteritems(ref):
+                            if __really_unused_name__xxxxx__ is __really_unused_name__342424__:
+                                names.append(name)
 
-        df = DataFrame(np.arange(0,9), columns=['count'])
-        df['group'] = 'b'
+                for name, __really_unused_name__xxxxx__ in compat.iteritems(globals()):
+                    if __really_unused_name__xxxxx__ is __really_unused_name__342424__:
+                        names.append(name)
 
-        # this technically need not raise SettingWithCopy if both are view (which is not
-        # generally guaranteed but is usually True
-        # however, this is in general not a good practice and we recommend using .loc
-        df.iloc[0:5]['group'] = 'a'
+                return set(names) - removals
 
-        """
+            # collect garbage
+            # if we don't have references, then we have a reassignment case
+            #    e.g. df = df.ix[....]; since the reference is gone
+            #    we can just copy and be done
 
-        if force or self.is_copy:
+            # otherwise we have chained indexing, raise and error
+            def error():
+                raise SettingWithCopyError("chained indexing detected, you can fix this ......")
 
-            value = config.get_option('mode.chained_assignment')
-            if value is None:
-                return
+            gc.collect(2)
+            if len(self._parent) > 1:
+                error()
 
-            # see if the copy is not actually refererd; if so, then disolve
-            # the copy weakref
-            try:
-                gc.collect(2)
-                if not gc.get_referents(self.is_copy()):
-                    self.is_copy = None
-                    return
-            except:
-                pass
+            p = self._parent[0]()
+            if p is not None:
+                names = get_names_for_obj(self)
```

Comment:
@jreback If the fragile `get_names_for_obj` is merely for prohibiting chained indexing, then perhaps we can get rid of it altogether? With this change, chained indexing will *always* fail (due to copy-on-write), so I don't think we actually need this anymore.